packages/framework-core/src/services/graphql/graphql-query-generator.ts

```diff
@@ -141,75 +132,137 @@ export class GraphQLQueryGenerator {
+        const primitiveType = this.typeInformer.getPrimitiveExtendedType(param.type)
+        let graphqlType: GraphQLScalarType
+        switch (primitiveType) {
+          case Boolean:
+            graphqlType = GraphQLBoolean
+            break
+          case String:
+            graphqlType = GraphQLString
+            break
+          case Number:
+            graphqlType = GraphQLFloat
+            break
+          default:
+            graphqlType = GraphQLJSONObject
+            break
+        }
+        propFilters.includes = {
+          ...propFilters.includes,
+          type: graphqlType,
+        }
+      })
+
       this.generatedFiltersByTypeName[filterName] = new GraphQLInputObjectType({
         name: filterName,
-        fields: {
-          operation: { type: new GraphQLNonNull(this.operationEnumFor(type)) },
-          values: { type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(graphQLValueType))) },
-        },
+        fields: propFilters,
       })
     }
     return this.generatedFiltersByTypeName[filterName]
   }
 
-  private operationEnumFor(type: AnyClass): GraphQLEnumType {
-    const operationEnumName = `${type.name}Operations`
-    if (!this.generatedOperationEnumsByTypeName[operationEnumName]) {
-      this.generatedOperationEnumsByTypeName[operationEnumName] = new GraphQLEnumType({
-        name: operationEnumName,
-        values: this.generateOperationEnumValuesFor(type),
-      })
+  private generateFilterFor(prop: PropertyMetadata): GraphQLInputObjectType | GraphQLScalarType {
+    const filterName = `${prop.typeInfo.name}PropertyFilter`
+
+    if (prop.typeInfo.type === undefined) return GraphQLJSONObject
+
+    const primitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
+    if (primitiveType === Array) return this.generateArrayFilterFor(prop)
+    if (!this.generatedFiltersByTypeName[filterName]) {
+      const graphQLPropType = this.typeInformer.getGraphQLTypeFor(primitiveType)
+      let fields: Thunk<GraphQLInputFieldConfigMap> = {}
+
+      if (!this.typeInformer.isPrimitiveType(graphQLPropType)) {
+        let nestedProperties: GraphQLInputFieldConfigMap = {}
+        const properties = getPropertiesMetadata(prop.typeInfo.type)
+        if (properties.length > 0) {
+          this.typeInformer.generateGraphQLTypeFromMetadata({ class: prop.typeInfo.type, properties })
+
+          for (const prop of properties) {
+            const property = { [prop.name]: { type: this.generateFilterFor(prop) } }
+            nestedProperties = { ...nestedProperties, ...property }
+          }
+        } else {
+          return GraphQLJSONObject
+        }
+        fields = () => ({
+          ...nestedProperties,
+          and: { type: new GraphQLList(this.generatedFiltersByTypeName[filterName]) },
+          or: { type: new GraphQLList(this.generatedFiltersByTypeName[filterName]) },
+          not: { type: this.generatedFiltersByTypeName[filterName] },
```

Comment:
Hahaha I don't blame you! We are setting it here **before** the value is set. Why is that possible? It's because using fields as a function instead of as an object, the graphql library understands that it's a recursive call and the value will be set later. 

There is not much info about it, but you can find an example when setting the `PersonType` on this example: https://graphql.org/graphql-js/type/#examples

Reply:
Gotcha! Thanks for explaining